Theme konnectd

LukasHirt opened the pull request:
Bring konnectd frontend, create ownCloud theme and generate assets.

![image](https://user-images.githubusercontent.com/25989331/96376119-5c2d7800-117d-11eb-9670-f3ae250cec87.png)

## Open tasks:
- [x] Changelog item
- [x] Replace placeholder logo with the ownCloud one when generating
- [x] Bring ownCloud favicon when generating

LukasHirt commented:
There are some parts of the frontend which are probably not required but were causing a lot of errors when removed. I'd like to go in a separate PR through them when there'd more time and do some followup cleanup.

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,69 @@
+.oc-button-primary:hover,
+.oc-button-primary:focus {
+  /* Needs to be important to overwrite material-ui */
+  background-color: #306db5 !important;
```
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [When to Use the `public` Folder](#when-to-use-the-public-folder)
+- [Using Global Variables](#using-global-variables)
+- [Adding Bootstrap](#adding-bootstrap)
+  - [Using a Custom Theme](#using-a-custom-theme)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/build/static/css/main.5634be8f.chunk.css:
```diff
@@ -0,0 +1,2 @@
+body,html{height:100%}body{margin:0;padding:0}#bg{position:fixed;top:0;bottom:0;left:0;right:0;background:#fff}#loader{position:absolute;bottom:6%;left:50%;font-weight:400;font-family:"Roboto","Helvetica","Arial",sans-serif;color:#666;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%)}#loader>a{color:#1e98d7}#font-preloader{position:absolute;top:-500px;font-weight:400;color:#ff ...
```
Codacy found an issue: [Expected newline after "}" (block-closing-brace-newline-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space before ">" (selector-combinator-space-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after ";" in a single-line declaration block (declaration-block-semicolon-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space before "}" of a single-line block (block-closing-brace-space-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after ">" (selector-combinator-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected newline after "," (selector-list-comma-newline-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/containers/Goodbye/Goodbyescreen.js:
```diff
@@ -0,0 +1,125 @@
+  }
+
+  render() {
+    const { classes, hello } = this.props;
```
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [npm test](#npm-test)
+  - [npm run build](#npm-run-build)
+  - [npm run eject](#npm-run-eject)
+- [Supported Language Features and Polyfills](#supported-language-features-and-polyfills)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [Adding Assets Outside of the Module System](#adding-assets-outside-of-the-module-system)
+  - [When to Use the `public` Folder](#when-to-use-the-public-folder)
+- [Using Global Variables](#using-global-variables)
+- [Adding Bootstrap](#adding-bootstrap)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/build/static/js/10.fe240f90.chunk.js:
```diff
@@ -0,0 +1,3 @@
+/*! For license information please see 10.fe240f90.chunk.js.LICENSE.txt */
+(this.webpackJsonpidentifier=this.webpackJsonpidentifier||[]).push([[10],{173:function(e,t,r){"use strict";Object.defineProperty(t,"__esModule",{value:!0}),t.default=void 0;var o={50:"#e8f5e9",100:"#c8e6c9",200:"#a5d6a7",300:"#81c784",400:"#66bb6a",500:"#4caf50",600:"#43a047",700:"#388e3c",800:"#2e7d32",900:"#1b5e20",A100:"#b9f6ca",A200:"#69f0ae",A400:"#00e676",A700:"#00c853"};t.default=o},340:fu ...
```
Codacy found an issue: [The numeric literal '.9' will have at different value at runtime.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/build/static/js/0.d0436ff0.chunk.js:
```diff
@@ -0,0 +1,3 @@
+/*! For license information please see 0.d0436ff0.chunk.js.LICENSE.txt */
+(this.webpackJsonpidentifier=this.webpackJsonpidentifier||[]).push([[0],{107:function(e,t,n){var r;!function(){"use strict";var n={}.hasOwnProperty;function o(){for(var e=[],t=0;t<arguments.length;t++){var r=arguments[t];if(r){var a=typeof r;if("string"===a||"number"===a)e.push(r);else if(Array.isArray(r)&&r.length){var i=o.apply(null,r);i&&e.push(i)}else if("object"===a)for(var c in r)n.call(r,c ...
```
Codacy found an issue: [The numeric literal '.01' will have at different value at runtime.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/fancy-background.css:
```diff
@@ -0,0 +1,36 @@
+#loader {
+  /* NOTE(longsleep): White here needed because of the background image */
+  color: white;
+  text-shadow: #000 0px 0px 1px;
```
Codacy found an issue: [Unexpected unit (length-zero-no-unit)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unexpected unit (length-zero-no-unit)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unexpected unit (length-zero-no-unit)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unexpected unit (length-zero-no-unit)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [npm start](#npm-start)
+  - [npm test](#npm-test)
+  - [npm run build](#npm-run-build)
+  - [npm run eject](#npm-run-eject)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,69 @@
+}
+
+.oc-mt-s {
+  margin-top: 10px !important;
```
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/build/static/js/identifier-app.cde47c8f.chunk.js:
```diff
@@ -0,0 +1,2 @@
+(this.webpackJsonpidentifier=this.webpackJsonpidentifier||[]).push([[4],{205:function(e){e.exports=JSON.parse('{"konnect.loading.error.headline":"Verbindung zum Server fehlgeschlagen","konnect.login.retryButton.label":"Wiederholen","konnect.scopeDescription.aliasBasic":"Zugriff auf Ihre grundlegenden Benutzerinformationen","konnect.scopeDescription.offlineAccess":"Dauerhaften Zugriff (l\xe4uft ni ...
```
Codacy found an issue: [The numeric literal '2e4' will have at different value at runtime.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Debugging in the Editor](#debugging-in-the-editor)
+- [Formatting Code Automatically](#formatting-code-automatically)
+- [Changing the Page `<title>`](#changing-the-page-title)
+- [Installing a Dependency](#installing-a-dependency)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+## Table of Contents
+
+- [Updating to New Releases](#updating-to-new-releases)
+- [Sending Feedback](#sending-feedback)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+
+## Table of Contents
+
+- [Updating to New Releases](#updating-to-new-releases)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+
+- [Updating to New Releases](#updating-to-new-releases)
+- [Sending Feedback](#sending-feedback)
+- [Folder Structure](#folder-structure)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Using the `public` Folder](#using-the-public-folder)
+  - [Changing the HTML](#changing-the-html)
+  - [Adding Assets Outside of the Module System](#adding-assets-outside-of-the-module-system)
+  - [When to Use the `public` Folder](#when-to-use-the-public-folder)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/build/static/js/runtime-main.670cdf97.js:
```diff
@@ -0,0 +1,2 @@
+!function(e){function r(r){for(var n,a,u=r[0],c=r[1],f=r[2],s=0,p=[];s<u.length;s++)a=u[s],Object.prototype.hasOwnProperty.call(o,a)&&o[a]&&p.push(o[a][0]),o[a]=0;for(n in c)Object.prototype.hasOwnProperty.call(c,n)&&(e[n]=c[n]);for(l&&l(r);p.length;)p.shift()();return i.push.apply(i,f||[]),t()}function t(){for(var e,r=0;r<i.length;r++){for(var t=i[r],n=!0,u=1;u<t.length;u++){var c=t[u];0!==o[c]& ...
```
Codacy found an issue: [The numeric literal '12e4' will have at different value at runtime.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Updating to New Releases](#updating-to-new-releases)
+- [Sending Feedback](#sending-feedback)
+- [Folder Structure](#folder-structure)
+- [Available Scripts](#available-scripts)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Code Splitting](#code-splitting)
+- [Adding a Stylesheet](#adding-a-stylesheet)
+- [Post-Processing CSS](#post-processing-css)
+- [Adding a CSS Preprocessor (Sass, Less etc.)](#adding-a-css-preprocessor-sass-less-etc)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Sending Feedback](#sending-feedback)
+- [Folder Structure](#folder-structure)
+- [Available Scripts](#available-scripts)
+  - [npm start](#npm-start)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,69 @@
+
+.oc-mt-s {
+  margin-top: 10px !important;
+}
```
Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Changing the Page `<title>`](#changing-the-page-title)
+- [Installing a Dependency](#installing-a-dependency)
+- [Importing a Component](#importing-a-component)
+- [Code Splitting](#code-splitting)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Formatting Code Automatically](#formatting-code-automatically)
+- [Changing the Page `<title>`](#changing-the-page-title)
+- [Installing a Dependency](#installing-a-dependency)
+- [Importing a Component](#importing-a-component)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/actions/login.js:
```diff
@@ -0,0 +1,291 @@
+export const ModeLogonUsernamePassword = '1';
+
+export function updateInput(name, value) {
+  return {
```
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Adding a Stylesheet](#adding-a-stylesheet)
+- [Post-Processing CSS](#post-processing-css)
+- [Adding a CSS Preprocessor (Sass, Less etc.)](#adding-a-css-preprocessor-sass-less-etc)
+- [Adding Images, Fonts, and Files](#adding-images-fonts-and-files)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/fancy-background.css:
```diff
@@ -0,0 +1,36 @@
+}
+
+/* NOTE(longsleep): This imports the inline image CSS as generated by make. */
+/* @import url(./images/loginscreen-bg.css); */
```
Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/components/Loading.js:
```diff
@@ -0,0 +1,93 @@
+
+class Loading extends React.PureComponent {
+  render() {
+    const { classes, error } = this.props;
```
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [Changing the HTML](#changing-the-html)
+  - [Adding Assets Outside of the Module System](#adding-assets-outside-of-the-module-system)
+  - [When to Use the `public` Folder](#when-to-use-the-public-folder)
+- [Using Global Variables](#using-global-variables)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Post-Processing CSS](#post-processing-css)
+- [Adding a CSS Preprocessor (Sass, Less etc.)](#adding-a-css-preprocessor-sass-less-etc)
+- [Adding Images, Fonts, and Files](#adding-images-fonts-and-files)
+- [Using the `public` Folder](#using-the-public-folder)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,69 @@
+/* Helpers */
+
+.oc-w-100 {
+  width: 100% !important;
```
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Adding a CSS Preprocessor (Sass, Less etc.)](#adding-a-css-preprocessor-sass-less-etc)
+- [Adding Images, Fonts, and Files](#adding-images-fonts-and-files)
+- [Using the `public` Folder](#using-the-public-folder)
+  - [Changing the HTML](#changing-the-html)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [npm run build](#npm-run-build)
+  - [npm run eject](#npm-run-eject)
+- [Supported Language Features and Polyfills](#supported-language-features-and-polyfills)
+- [Syntax Highlighting in the Editor](#syntax-highlighting-in-the-editor)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Supported Language Features and Polyfills](#supported-language-features-and-polyfills)
+- [Syntax Highlighting in the Editor](#syntax-highlighting-in-the-editor)
+- [Displaying Lint Output in the Editor](#displaying-lint-output-in-the-editor)
+- [Debugging in the Editor](#debugging-in-the-editor)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,69 @@
+
+.oc-button-primary {
+  /* Needs to be important to overwrite material-ui */
+  background-color: #4e85c8 !important;
```
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Use of !important](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Importing a Component](#importing-a-component)
+- [Code Splitting](#code-splitting)
+- [Adding a Stylesheet](#adding-a-stylesheet)
+- [Post-Processing CSS](#post-processing-css)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+  - [npm run eject](#npm-run-eject)
+- [Supported Language Features and Polyfills](#supported-language-features-and-polyfills)
+- [Syntax Highlighting in the Editor](#syntax-highlighting-in-the-editor)
+- [Displaying Lint Output in the Editor](#displaying-lint-output-in-the-editor)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Adding Images, Fonts, and Files](#adding-images-fonts-and-files)
+- [Using the `public` Folder](#using-the-public-folder)
+  - [Changing the HTML](#changing-the-html)
+  - [Adding Assets Outside of the Module System](#adding-assets-outside-of-the-module-system)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/README.react.md:
```diff
@@ -0,0 +1,2164 @@
+- [Installing a Dependency](#installing-a-dependency)
+- [Importing a Component](#importing-a-component)
+- [Code Splitting](#code-splitting)
+- [Adding a Stylesheet](#adding-a-stylesheet)
```
Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui_config/webpack.config.js:
```diff
@@ -0,0 +1,676 @@
+// makes for a smoother build process.
+const shouldInlineRuntimeChunk = process.env.INLINE_RUNTIME_CHUNK !== 'false';
+
+const imageInlineSizeLimit = parseInt(
```
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Always provide a base when using parseInt() functions](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/actions/common.js:
```diff
@@ -0,0 +1,133 @@
+import * as types from './types';
+
+export function receiveError(error) {
+  return {
```
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/actions/common.js:
```diff
@@ -0,0 +1,133 @@
+}
+
+export function receiveHello(hello) {
+  const { success, username, displayName } = hello;
```
Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

LukasHirt commented:
@micbar Have you got some idea why this line is not respected by codacy https://github.com/owncloud/ocis/pull/698/files#diff-3404650b2859b064f8b0527d200936d6b306cf8e11437f0051b38af5aeaab1c2R4 and the error in webpack config is still showing up?

micbar commented:
@LukasHirt this file is no longer used.

Use the top level one and adapt the paths

micbar commented:
same for the `.gitignore`

kulmann commented:
@LukasHirt please rebase to current master. A `grpc-ecosystem` package that we are using in some Makefiles has been renamed => already adjusted on master.

LukasHirt commented:
> please rebase to current master. A grpc-ecosystem package that we are using in some Makefiles has been renamed => already adjusted on master.

Done 😉

LukasHirt commented:
> Use the top level one and adapt the paths

Still getting the error even though I moved it in there 😞

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+  background-color: #042047;
+  border: 1px solid rgba(78,133,200,.8);
+  border-radius: 3px;
+  color: rgba(255,255,255,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+
+.oc-input {
+  background-color: #042047;
+  border: 1px solid rgba(78,133,200,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+}
+
+.oc-input::placeholder {
+  color: rgba(78,133,200,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+  background-color: #042047;
+  border: 1px solid rgba(78,133,200,.8);
+  border-radius: 3px;
+  color: rgba(255,255,255,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+}
+
+.oc-input::placeholder {
+  color: rgba(78,133,200,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,88 @@
+
+.oc-input {
+  background-color: #042047;
+  border: 1px solid rgba(78,133,200,.8);
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

micbar commented on .codacy.yml:
```diff
@@ -7,7 +7,8 @@ exclude_paths:
   - '**/pkg/proto/**'
-  - 'konnectd/assets/identifier/**'
+  - 'konnectd/ui_config/**/*'
+  - 'konnectd/scripts/**/*'
```
remove the last `/*` to include also the files in the `ui_config` folder.
Your glob patttern only selects files inside of subfolders of `ui_config`.

See https://globster.xyz/?q=%2Fmyapp%2Fsrc%2F**&f=%2Fmyapp%2Freadme.md%2C%2Fmyapp%2Fconfig%2Fstaging.js%2C%2Fmyapp%2Fconfig%2Fproduction.js%2C%2Fmyapp%2Fsrc%2Fservices%2Futils.js%2C%2Fmyapp%2Fsrc%2Fservices%2Ftimezone.ts%2C%2Fmyapp%2Fsrc%2Fcontrollers%2Fhealth.js%2C%2Fmyapp%2Fsrc%2Fcontrollers%2Fuser.module.ts%2C%2Fmyapp%2Fsrc%2FREADME.md%2C%2Fmyapp%2Fassets%2Flogo.png%2C%2Fmyapp%2Fassets%2Flogo_small.png%2C%2Fmyapp%2Fassets%2Ffavicon.ico

kulmann reviewed (changes requested):
Noticed some things:
- file `konnectd/ui/src/images/owncloud-logo.svg` is part of this PR and should be removed or some placeholder image, right?
- There are some files from the original kopano theme (augmenting teamwork, etc) which could be removed
- The font size of the input fields is too small. It should be 1rem, but is 0.8...rem. With the current size it looks weird coming from the login form with tiny font size to ownCloud Web with quite big font size.

LukasHirt commented:
> file konnectd/ui/src/images/owncloud-logo.svg is part of this PR and should be removed or some placeholder image, right?

Right, thanks!

> There are some files from the original kopano theme (augmenting teamwork, etc) which could be removed

I removed leftover images for now but regarding other files is this comment:

`There are some parts of the frontend which are probably not required but were causing a lot of errors when removed. I'd like to go in a separate PR through them when there'd more time and do some followup cleanup.`

> The font size of the input fields is too small. It should be 1rem, but is 0.8...rem. With the current size it looks weird coming from the login form with tiny font size to ownCloud Web with quite big font size.

Per request from the marketing team, I have adjusted this according to the ownCloud website form inputs where it is the same font size.

kulmann commented:
> > The font size of the input fields is too small. It should be 1rem, but is 0.8...rem. With the current size it looks weird coming from the login form with tiny font size to ownCloud Web with quite big font size.
> 
> Per request from the marketing team, I have adjusted this according to the ownCloud website form inputs where it is the same font size.

Hm... I don't agree. 😅 Styling the login page should be consistent with phoenix, not the website. 👀  We're going back and forth between phoenix and the login page, the website is never part of this process.

refs commented on konnectd/ui/src/containers/Goodbye/Goodbyescreen.js:
```diff
@@ -0,0 +1,125 @@
+  }
+}
+
+Goodbyescreen.propTypes = {
```
Is using TS an option? https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html#moving-to-typescript-files

LukasHirt replied:
I'd very much love that! Not sure if I can introduce in one PR not only react but TS as well though 🙈

refs commented on konnectd/ui/src/containers/Login/Chooseaccount.js:
```diff
@@ -0,0 +1,150 @@
+});
+
+class Chooseaccount extends React.PureComponent {
+  componentDidMount() {
```
I believe a [functional component + an effect hook](https://reactjs.org/docs/hooks-overview.html#effect-hook) would allow you to trash ugly js classes in favor of pure magnificent functions <3

LukasHirt replied:
Thanks for the tip on effect hook 👼

refs commented on konnectd/ui/src/index.js:
```diff
@@ -0,0 +1,13 @@
+console.info(`Kopano Identifier build version: ${version.build}`); // eslint-disable-line no-console
+console.info(`Kopano Kpop build version: ${kpop.build}`); // eslint-disable-line no-console
+
+// NOTE(longsleep): Load async, this enables code splitting via Webpack.
```
want to leave this comment?

refs reviewed (approved):
First of all, huge PR 👏 I took a look mostly at the react components, leaving build stuff out, and just a matter of consistency and a suggestion for using TypeScript 🤓 . Haven't tried the PR locally, intending to do so later today.

LukasHirt commented on konnectd/ui/src/components/Loading.js:
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
```
This would create extra dependency on Vue, wouldn't it? At least when looking at https://www.npmjs.com/package/react-vuex that's the feeling I am getting

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,104 @@
+
+.oc-progress {
+  /* Needs to be important to overwrite material-ui */
+  background-color: rgba(78,133,200,.8) !important;
```
Codacy found an issue: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders replied:
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,104 @@
+
+.oc-light {
+  color: #ffffff !important;
+}
```
Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,104 @@
+}
+
+.oc-light {
+  color: #ffffff !important;
```
Codacy found an issue: [Expected "#ffffff" to be "#fff" (color-hex-length)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

LukasHirt commented:
Addressed your comments @refs and increased font size and changed the height of inputs to reflect the other ones in ocis-web @kulmann Thank you both for reviews! I've rebased the PR also so hopefully the random API tests failures will be gone.

ownclouders commented on konnectd/ui/src/app.css:
```diff
@@ -0,0 +1,104 @@
+
+.oc-progress {
+  /* Needs to be important to overwrite material-ui */
+  background-color: rgba(78, 133, 200, .8) !important;
```
Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6337760)

LukasHirt commented on konnectd/ui/src/components/Loading.js:
```diff
@@ -0,0 +1,93 @@
+  }
+});
+
+class Loading extends React.PureComponent {
```
Turned into FC 🚀

kulmann reviewed (approved):
🚀

kulmann commented:
love it!
